### PR TITLE
feat(ST-39): registration form for student and validation added

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Registration form for student is added.
- Form validation is added.
- Made responsive for mobile device.
- Background color is adjusted for the app.

## Associated Ticket Link(s)

- https://trello.com/c/iNgiAXEM

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/120fb37b-0cf0-4f07-935c-8de3a0b9849f

### CODE REVIEW

File: modules/Registration/components/studentRegistrationSchema.ts
Comment (ishmam-iqbal-sazim) [inline modules/Registration/components/studentRegistrationSchema.ts:1]: This should be in `StudentRegistrationForm.helpers.ts` file.

File: modules/Registration/components/StudentRegistrationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+export function StudentRegistrationForm({ onSubmit }: IStudentRegistrationFormProps) {
+  const {
+    control,
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+    reset,
+  } = useForm<StudentRegistrationFormData>({
+    resolver: zodResolver(studentRegistrationSchema),
```
Comment (ishmam-iqbal-sazim) [inline modules/Registration/components/StudentRegistrationForm.tsx:24]: Export the resolver from `StudentRegistrationForm.helpers.ts` file and import it here like so: 
```suggestion
    resolver: StudentRegistrationSchemaResolver,
```

File: pages/registration/student/index.page.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+import { StudentRegistrationContainer } from "@/modules/Registration/containers/StudentRegistration";
```
Comment (ishmam-iqbal-sazim) [inline pages/registration/student/index.page.tsx:3]:
```suggestion
import { StudentRegistrationContainer } from "@/modules/Registration/containers/StudentRegistrationContainer";
```

Let''s be consistent with the naming

File: modules/Registration/components/studentRegistrationSchema.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  degreeType: undefined,
+  degreeName: undefined,
+  semesterYear: undefined,
+};
+
+export const validateStudentRegistration = (data: StudentRegistrationFormData) => {
+  const result = studentRegistrationSchema.safeParse(data);
+  if (result.success) return { values: result.data, errors: {} };
+  return { values: {}, errors: result.error.flatten().fieldErrors };
+};
```
Comment (ishmam-iqbal-sazim) [inline modules/Registration/components/studentRegistrationSchema.ts:67]: I am not sure what this method is intended for.

File: modules/Registration/components/student/StudentRegistrationForm.tsx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+              <Controller
+                name="class"
+                control={control}
+                render={({ field }) => (
+                  <Select
+                    label="Class"
+                    placeholder="Select class"
+                    classNames={{ input: styles["input"], label: styles["label"] }}
+                    data={[
+                      { value: "1", label: "Class 1" },
```
Comment (ragib-hossain-sazim) [inline modules/Registration/components/student/StudentRegistrationForm.tsx:199]: Could you please move this options to a constant file?

File: modules/Registration/components/student/StudentRegistrationForm.tsx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+                <Controller
+                  name="degreeName"
+                  control={control}
+                  render={({ field }) => (
+                    <Select
+                      label="Degree Name"
+                      placeholder="Select degree name"
+                      classNames={{ input: styles["input"], label: styles["label"] }}
+                      data={[
+                        { value: "computerScience", label: "Computer Science" },
```
Comment (ragib-hossain-sazim) [inline modules/Registration/components/student/StudentRegistrationForm.tsx:224]: Same comment as above

File: modules/Registration/components/student/StudentRegistrationForm.tsx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+                        color: "#333",
+                      },
+                      item: {
+                        "&[data-selected]": {
+                          color: "#333",
+                        },
+                      },
+                    }}
+                    data={[
+                      { value: "School", label: "School" },
```
Comment (ragib-hossain-sazim) [inline modules/Registration/components/student/StudentRegistrationForm.tsx:136]: Please move this options to a constant file

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments

File: modules/Registration/components/student/StudentRegistrationForm.helpers.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    educationLevel: educationLevelEnum.optional().refine((val) => val !== undefined, {
+      message: "Education level is required",
+    }),
+    password: z.string().min(6, "Password must be at least 6 characters"),
+    confirmPassword: z.string(),
+    englishBanglaMedium: z.string().optional(),
+    class: z.string().optional(),
+    degreeType: z.enum(["Bachelors", "Masters"]).optional(),
+    degreeName: z.string().optional(),
+    semesterYear: z.string().optional(),
```
Comment (ishmam-iqbal-sazim) [inline modules/Registration/components/student/StudentRegistrationForm.helpers.ts:23]: Please look into the usage of superRefine and how to make these fields required based on educationLevel field.

Can pair with Faysal bhai as he has implemented this.

Comment (akib-raihan-sazim) [reply modules/Registration/components/student/StudentRegistrationForm.helpers.ts:23]: [
education_level_error
](url)

File: modules/Registration/components/student/StudentRegistrationForm.helpers.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  address: string;
+  educationLevel?: "School" | "College" | "University";
+  password: string;
+  confirmPassword: string;
+  englishBanglaMedium?: string;
+  class?: string;
+  degreeType?: "Bachelors" | "Masters";
+  degreeName?: string;
+  semesterYear?: string;
+};
```
Comment (ishmam-iqbal-sazim) [inline modules/Registration/components/student/StudentRegistrationForm.helpers.ts:88]: This should be in the types file

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Pull request merged**